diff --git a/mobile/android/android-components/components/feature/search/src/main/java/mozilla/components/feature/search/storage/SearchEngineReader.kt b/mobile/android/android-components/components/feature/search/src/main/java/mozilla/components/feature/search/storage/SearchEngineReader.kt
index ae405c6a94..1e06aeeb48 100644
--- a/mobile/android/android-components/components/feature/search/src/main/java/mozilla/components/feature/search/storage/SearchEngineReader.kt
+++ b/mobile/android/android-components/components/feature/search/src/main/java/mozilla/components/feature/search/storage/SearchEngineReader.kt
@@ -38,16 +38,17 @@ private const val MAX_SIZE = 32
 // List of general search engine ids, taken from
 // https://searchfox.org/mozilla-central/rev/ef0aa879e94534ffd067a3748d034540a9fc10b0/toolkit/components/search/SearchUtils.sys.mjs#200
 internal val GENERAL_SEARCH_ENGINE_IDS = setOf(
-    GOOGLE_ID,
+//    GOOGLE_ID,
     "ddg",
-    "bing",
-    "baidu",
-    "ecosia",
-    "qwant",
-    "yahoo-jp",
-    "seznam-cz",
-    "coccoc",
-    "baidu",
+    "ddgnoai",
+    "ddghtml",
+    "ddglite",
+    "mojeek",
+    "leta-brave",
+    "leta-google",
+    "startpage",
+    "startpage-eu",
+    "no-search",
 )
 
 /**
