diff --git a/mobile/android/android-components/build.gradle b/mobile/android/android-components/build.gradle
index f7336a3e78..19861a695e 100644
--- a/mobile/android/android-components/build.gradle
+++ b/mobile/android/android-components/build.gradle
@@ -9,14 +9,9 @@ import static org.gradle.api.tasks.testing.TestResult.ResultType
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
@@ -33,14 +28,9 @@ plugins {
 
 allprojects {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
 
         maven {
             url = "${gradle.mozconfig.topobjdir}/gradle/maven"
diff --git a/mobile/android/android-components/components/browser/engine-gecko/build.gradle b/mobile/android/android-components/components/browser/engine-gecko/build.gradle
index 6bbc5b3ed8..e009561361 100644
--- a/mobile/android/android-components/components/browser/engine-gecko/build.gradle
+++ b/mobile/android/android-components/components/browser/engine-gecko/build.gradle
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff --git a/mobile/android/android-components/components/feature/fxsuggest/build.gradle b/mobile/android/android-components/components/feature/fxsuggest/build.gradle
index 1fdc583cf7..933b4ec8d0 100644
--- a/mobile/android/android-components/components/feature/fxsuggest/build.gradle
+++ b/mobile/android/android-components/components/feature/fxsuggest/build.gradle
@@ -6,9 +6,7 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 buildscript {
     repositories {
-        maven {
-            url = "https://maven.mozilla.org/maven2"
-        }
+        mavenLocal()
     }
 
     dependencies {
diff --git a/mobile/android/android-components/components/lib/crash/build.gradle b/mobile/android/android-components/components/lib/crash/build.gradle
index fe86bfc252..68713e57d7 100644
--- a/mobile/android/android-components/components/lib/crash/build.gradle
+++ b/mobile/android/android-components/components/lib/crash/build.gradle
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff --git a/mobile/android/android-components/components/service/mars/build.gradle b/mobile/android/android-components/components/service/mars/build.gradle
index 79a9aea939..7841f131f1 100644
--- a/mobile/android/android-components/components/service/mars/build.gradle
+++ b/mobile/android/android-components/components/service/mars/build.gradle
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff --git a/mobile/android/android-components/components/service/nimbus/build.gradle b/mobile/android/android-components/components/service/nimbus/build.gradle
index 7f79cbdbd6..0a4b0c3a72 100644
--- a/mobile/android/android-components/components/service/nimbus/build.gradle
+++ b/mobile/android/android-components/components/service/nimbus/build.gradle
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff --git a/mobile/android/android-components/components/service/pocket/build.gradle b/mobile/android/android-components/components/service/pocket/build.gradle
index 0b19b3e03d..5eb26455bf 100644
--- a/mobile/android/android-components/components/service/pocket/build.gradle
+++ b/mobile/android/android-components/components/service/pocket/build.gradle
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff --git a/mobile/android/android-components/components/support/remotesettings/build.gradle b/mobile/android/android-components/components/support/remotesettings/build.gradle
index bdae594eb0..17a61a5dea 100644
--- a/mobile/android/android-components/components/support/remotesettings/build.gradle
+++ b/mobile/android/android-components/components/support/remotesettings/build.gradle
@@ -3,14 +3,9 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
     dependencies {
         classpath  libs.plugin.serialization
diff --git a/mobile/android/android-components/plugins/config/build.gradle b/mobile/android/android-components/plugins/config/build.gradle
index 3728558208..8cb258a793 100644
--- a/mobile/android/android-components/plugins/config/build.gradle
+++ b/mobile/android/android-components/plugins/config/build.gradle
@@ -9,14 +9,9 @@ plugins {
 buildDir = "${gradle.mozconfig.topobjdir}/gradle/build/mobile/android/android-components/plugins/config"
 
 repositories {
-    gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-        maven {
-            url = repository
-            if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                allowInsecureProtocol = true
-            }
-        }
-    }
+    mavenLocal()
+    maven { url "https://plugins.gradle.org/m2/" }
+    google()
 }
 
 gradlePlugin {
diff --git a/mobile/android/android-components/plugins/config/settings.gradle b/mobile/android/android-components/plugins/config/settings.gradle
index b826d02d17..8580140364 100644
--- a/mobile/android/android-components/plugins/config/settings.gradle
+++ b/mobile/android/android-components/plugins/config/settings.gradle
@@ -11,14 +11,9 @@ pluginManagement {
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 }
 
diff --git a/mobile/android/android-components/plugins/dependencies/build.gradle b/mobile/android/android-components/plugins/dependencies/build.gradle
index 9a79d65de7..27df676741 100644
--- a/mobile/android/android-components/plugins/dependencies/build.gradle
+++ b/mobile/android/android-components/plugins/dependencies/build.gradle
@@ -9,14 +9,9 @@ plugins {
 buildDir = "${gradle.mozconfig.topobjdir}/gradle/build/mobile/android/android-components/plugins/dependencies"
 
 repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
 }
 
 gradlePlugin {
diff --git a/mobile/android/android-components/plugins/dependencies/settings.gradle b/mobile/android/android-components/plugins/dependencies/settings.gradle
index b826d02d17..8580140364 100644
--- a/mobile/android/android-components/plugins/dependencies/settings.gradle
+++ b/mobile/android/android-components/plugins/dependencies/settings.gradle
@@ -11,14 +11,9 @@ pluginManagement {
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 }
 
diff --git a/mobile/android/android-components/plugins/publicsuffixlist/build.gradle b/mobile/android/android-components/plugins/publicsuffixlist/build.gradle
index b33fd9b5e1..2059e733d0 100644
--- a/mobile/android/android-components/plugins/publicsuffixlist/build.gradle
+++ b/mobile/android/android-components/plugins/publicsuffixlist/build.gradle
@@ -9,14 +9,9 @@ plugins {
 buildDir = "${gradle.mozconfig.topobjdir}/gradle/build/mobile/android/android-components/plugins/publicsuffixlist"
 
 repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
 }
 
 dependencies {
diff --git a/mobile/android/android-components/plugins/publicsuffixlist/settings.gradle b/mobile/android/android-components/plugins/publicsuffixlist/settings.gradle
index 84ca151cdc..a03568e09c 100644
--- a/mobile/android/android-components/plugins/publicsuffixlist/settings.gradle
+++ b/mobile/android/android-components/plugins/publicsuffixlist/settings.gradle
@@ -12,14 +12,9 @@ pluginManagement {
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     includeBuild("../dependencies")
diff --git a/mobile/android/android-components/settings.gradle b/mobile/android/android-components/settings.gradle
index 19dc9e9ccf..6adb734ad5 100644
--- a/mobile/android/android-components/settings.gradle
+++ b/mobile/android/android-components/settings.gradle
@@ -10,14 +10,9 @@ pluginManagement {
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     includeBuild("plugins/dependencies")
