diff --git a/mobile/android/android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/preparer/TippyTopIconPreparer.kt b/mobile/android/android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/preparer/TippyTopIconPreparer.kt
index 3763da85ad..ebcb11af85 100644
--- a/mobile/android/android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/preparer/TippyTopIconPreparer.kt
+++ b/mobile/android/android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/preparer/TippyTopIconPreparer.kt
@@ -60,7 +60,7 @@ class TippyTopIconPreparer(
                 type = IconRequest.Resource.Type.TIPPY_TOP,
             )
 
-            request.copy(resources = request.resources + resource)
+            request
         } else {
             request
         }
