diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 0bd15b53a6..465d524a37 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -289,7 +289,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      * Indicates if the top sites homepage section settings should be visible
      */
     val showHomepageTopSitesSectionToggle: Boolean
-        get() = !overrideUserSpecifiedHomepageSections || enableHomepageSearchBar
+        get() = false
 
     /**
      * Indicates if the recent tabs homepage section settings should be visible
@@ -2009,7 +2009,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var showContileFeature by booleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_contile),
-        default = true,
+        default = false,
     )
 
     /**
@@ -2673,7 +2673,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     var suppressSponsoredTopSitesEnabled by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_suppress_sponsored_tiles),
         featureFlag = true,
-        default = { FxNimbus.features.suppressSponsoredTopSites.value().enabled },
+        default = { true },
     )
 
     /**
