diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/MarketingPageRemovalSupport.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/MarketingPageRemovalSupport.kt
index 9e88fb54ae..5b94890932 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/MarketingPageRemovalSupport.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/MarketingPageRemovalSupport.kt
@@ -72,10 +72,15 @@ class MarketingPageRemovalSupport(
 
 internal fun MutableList<OnboardingPageUiData>.removeIfPageNotReached(index: Int) {
     val marketingIndex = indexOfFirst { it.type == OnboardingPageUiData.Type.MARKETING_DATA }
+    val syncIndex = indexOfFirst { it.type == OnboardingPageUiData.Type.SYNC_SIGN_IN }
 
     if (index < marketingIndex) {
         removeAt(marketingIndex)
     }
+
+    if (index < syncIndex) {
+        removeAt(syncIndex)
+    }
 }
 
 internal fun SharedPreferences.flowScopedBooleanPreference(
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SuggestionsStringsProvider.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SuggestionsStringsProvider.kt
index 351b05e484..fe1d586571 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SuggestionsStringsProvider.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SuggestionsStringsProvider.kt
@@ -128,15 +128,15 @@ class DefaultSuggestionsStringsProvider(
 
         var searchEngineName = defaultSearchEngine?.name
         if (!searchEngineName.isNullOrEmpty()) {
-            searchEngineName = when (searchEngineName) {
-                GOOGLE_SEARCH_ENGINE_NAME -> context.getString(
-                    R.string.google_search_engine_suggestion_header,
-                )
-                else -> context.getString(
+            searchEngineName =
+//                GOOGLE_SEARCH_ENGINE_NAME -> context.getString(
+//                    R.string.google_search_engine_suggestion_header,
+//                )
+                context.getString(
                     R.string.other_default_search_engine_suggestion_header,
                     searchEngineName,
                 )
-            }
+//            }
         }
         return searchEngineName
     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt
index b45fdee550..45dbba4891 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt
@@ -206,11 +206,11 @@ class AboutFragment(
             AboutPageItem(
                 AboutItem.ExternalLink(
                     SUPPORT,
-                    SupportUtils.getSumoURLForTopic(context, SupportUtils.SumoTopic.HELP),
+                    "https://gitlab.com/ironfox-oss/IronFox/-/issues",
                 ),
                 getString(R.string.about_support),
             ),
-            AboutPageItem(
+/*            AboutPageItem(
                 AboutItem.Crashes,
                 getString(R.string.about_crashes),
             ),
@@ -227,7 +227,7 @@ class AboutFragment(
                     SupportUtils.getSumoURLForTopic(context, SupportUtils.SumoTopic.YOUR_RIGHTS),
                 ),
                 getString(R.string.about_know_your_rights),
-            ),
+            ),*/
             AboutPageItem(
                 AboutItem.ExternalLink(LICENSING_INFO, ABOUT_LICENSE_URL),
                 getString(R.string.about_licensing_information),
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
index eb55c6630d..e8321e4fe2 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
@@ -84,7 +84,7 @@ class OnboardingFragment : Fragment() {
             pagesToDisplay(
                 showDefaultBrowserPage = isNotDefaultBrowser(this) && !isDefaultBrowserPromptSupported(),
                 showNotificationPage = canShowNotificationPage(),
-                showAddWidgetPage = canShowAddSearchWidgetPrompt(AppWidgetManager.getInstance(activity)),
+                showAddWidgetPage = false,
             ).toMutableList()
         }
     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
index 795fcd7b7e..1b4d2eef78 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingMapper.kt
@@ -32,7 +32,7 @@ internal fun Collection<OnboardingCardData>.toPageUiData(
         .sortedBy { it.ordering }
         .mapIndexed { index, onboardingCardData ->
             // only first onboarding card shows privacy caption
-            onboardingCardData.toPageUiData(if (index == 0) privacyCaption else null)
+            onboardingCardData.toPageUiData(null)
         }
 }
 
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
index 92a8a9a36d..c67e70ba65 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/TermsOfServiceOnboardingPage.kt
@@ -135,22 +135,22 @@ private fun BodyText(
             modifier = Modifier.padding(horizontal = 10.dp),
         ) {
             val lineOneState = LinkTextState(
-                text = it.lineOneLinkText,
+                text = "",
                 url = it.lineOneLinkUrl,
-                onClick = eventHandler::onTermsOfServiceLinkClicked,
+                onClick = {},
             )
             val lineTwoState = LinkTextState(
-                text = it.lineTwoLinkText,
+                text = "",
                 url = it.lineTwoLinkUrl,
-                onClick = eventHandler::onPrivacyNoticeLinkClicked,
+                onClick = {},
             )
             val lineThreeState = LinkTextState(
-                text = it.lineThreeLinkText,
+                text = "",
                 url = "",
-                onClick = { _ -> eventHandler.onManagePrivacyPreferencesLinkClicked() },
+                onClick = {},
             )
             LinkText(
-                text = it.lineOneText.updateFirstPlaceholder(it.lineOneLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineOneState,
                 ),
@@ -162,7 +162,7 @@ private fun BodyText(
             )
 
             LinkText(
-                text = it.lineTwoText.updateFirstPlaceholder(it.lineTwoLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineTwoState,
                 ),
@@ -174,7 +174,7 @@ private fun BodyText(
             )
 
             LinkText(
-                text = it.lineThreeText.updateFirstPlaceholder(it.lineThreeLinkText),
+                text = "",
                 linkTextStates = listOf(
                     lineThreeState,
                 ),
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index f0eeec863d..8906b4fce8 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -549,11 +549,11 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = false,
     )
 
-    var hasAcceptedTermsOfService by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_terms_accepted),
-        default = false,
-        persistDefaultIfNotExists = true,
-    )
+    var hasAcceptedTermsOfService = true
+//        appContext.getPreferenceKey(R.string.pref_key_terms_accepted),
+//        default = false,
+//        persistDefaultIfNotExists = true,
+//    )
 
     /**
      * Timestamp in milliseconds when the terms of use prompt was last shown to the user.
@@ -631,7 +631,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var shouldShowPrivacyPopWindow by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_privacy_pop_window),
-        default = true,
+        default = false,
     )
 
     var shouldUseLightTheme by booleanPreference(
