diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 0bd15b53a6..f9959c9255 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -190,7 +190,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         } else {
             preferences.getBoolean(
                 appContext.getPreferenceKey(R.string.pref_key_customization_bookmarks),
-                homescreenSections[HomeScreenSection.BOOKMARKS] == true,
+                false,
             )
         }
 
@@ -203,7 +203,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         } else {
             preferences.getBoolean(
                 appContext.getPreferenceKey(R.string.pref_key_recent_tabs),
-                homescreenSections[HomeScreenSection.JUMP_BACK_IN] == true,
+                false,
             )
         }
         set(value) {
@@ -239,7 +239,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         } else {
             preferences.getBoolean(
                 appContext.getPreferenceKey(R.string.pref_key_history_metadata_feature),
-                homescreenSections[HomeScreenSection.RECENT_EXPLORATIONS] == true,
+                false,
             )
         }
         set(value) {
@@ -257,8 +257,10 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     /**
      * Indicates whether or not the "Collections" section should be shown on the home screen.
      */
-    val collections: Boolean
-        get() = FxNimbus.features.homescreen.value().sectionsEnabled[HomeScreenSection.COLLECTIONS] == true
+    var collections: Boolean by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_collections_enabled),
+        default = true,
+    )
 
     /**
      * Indicates whether or not the homepage header should be shown.
@@ -314,7 +316,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     val overrideUserSpecifiedHomepageSections by lazyFeatureFlagPreference(
         appContext.getPreferenceKey(R.string.pref_key_override_user_specified_homepage_sections),
-        featureFlag = true,
+        featureFlag = false,
         default = { FxNimbus.features.overrideUserSpecifiedHomepageSections.value().enabled },
     )
 
@@ -524,7 +526,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var showCollectionsPlaceholderOnHome by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_show_collections_placeholder_home),
-        default = true,
+        default = false,
     )
 
     val isCrashReportingEnabled: Boolean
@@ -1391,7 +1393,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var defaultTopSitesAdded by booleanPreference(
         appContext.getPreferenceKey(R.string.default_top_sites_added),
-        default = false,
+        default = true,
     )
 
     var shouldShowSearchSuggestionsInPrivate by booleanPreference(
