diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
index 5340aa5258..1cc4e777f8 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
@@ -361,7 +361,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
                         // new search suggestions. The worker requires us to have called
                         // `GlobalFxSuggestDependencyProvider.initialize`, which we did before
                         // scheduling these tasks. When disabled we stop the periodic work.
-                        if (settings().enableFxSuggest) {
+                        if (settings().enableFxSuggest && settings().showNonSponsoredSuggestions) {
                             components.fxSuggest.ingestionScheduler.startPeriodicIngestion()
                         } else {
                             components.fxSuggest.ingestionScheduler.stopPeriodicIngestion()
@@ -452,7 +452,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
         queueStorageMaintenance()
         queueNimbusFetchInForeground()
         queueDownloadWallpapers()
-        if (settings().enableFxSuggest) {
+        if (settings().enableFxSuggest && settings().showNonSponsoredSuggestions) {
             queueSuggestIngest()
         }
         queueCollectProcessExitInfo()
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SearchSuggestionsProvidersBuilder.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SearchSuggestionsProvidersBuilder.kt
index b3e7086b89..ba7fbb3c7e 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SearchSuggestionsProvidersBuilder.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SearchSuggestionsProvidersBuilder.kt
@@ -81,7 +81,7 @@ class SearchSuggestionsProvidersBuilder(
                 components.core.icons,
                 engineForSpeculativeConnects,
                 showEditSuggestion = false,
-                suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//                suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             )
 
         defaultCombinedHistoryProvider =
@@ -93,7 +93,7 @@ class SearchSuggestionsProvidersBuilder(
                 engine = engineForSpeculativeConnects,
                 maxNumberOfSuggestions = METADATA_SUGGESTION_LIMIT,
                 showEditSuggestion = false,
-                suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//                suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             )
 
         val searchBitmap = suggestionIconProvider.getSearchIconBitmap()
@@ -319,7 +319,7 @@ class SearchSuggestionsProvidersBuilder(
                     engine = engineForSpeculativeConnects,
                     maxNumberOfSuggestions = METADATA_SUGGESTION_LIMIT,
                     showEditSuggestion = false,
-                    suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//                    suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
                     resultsUriFilter = filter::shouldIncludeUri,
                 )
             } else {
@@ -334,7 +334,7 @@ class SearchSuggestionsProvidersBuilder(
                     engine = engineForSpeculativeConnects,
                     maxNumberOfSuggestions = METADATA_SUGGESTION_LIMIT,
                     showEditSuggestion = false,
-                    suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//                    suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
                     resultsUriFilter = filter::shouldIncludeUri,
                 )
             } else {
@@ -446,7 +446,7 @@ class SearchSuggestionsProvidersBuilder(
                 suggestionIconProvider.getMobileIconDrawable(),
                 suggestionIconProvider.getTabletIconDrawable(),
             ),
-            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             resultsUrlFilter = filter?.let { it::shouldIncludeUrl },
         )
     }
@@ -468,7 +468,7 @@ class SearchSuggestionsProvidersBuilder(
             components.core.icons,
             suggestionIconProvider.getLocalTabIconDrawable(),
             excludeSelectedSession = !includeSelectedTab,
-            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             switchToTabDescription = suggestionsStringsProvider.getSwitchToTabDescriptionString(),
             resultsUriFilter = filter?.let { it::shouldIncludeUri },
         )
@@ -492,7 +492,7 @@ class SearchSuggestionsProvidersBuilder(
             indicatorIcon = suggestionIconProvider.getBookmarkIconDrawable(),
             engine = engineForSpeculativeConnects,
             showEditSuggestion = false,
-            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             resultsUriFilter = filter?.let { it::shouldIncludeUri },
         )
     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
index ce4b42ef4a..d5f03c9e2a 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
@@ -39,7 +39,7 @@ class SearchEngineFragment : PreferenceFragmentCompat() {
         )
 
         requirePreference<SwitchPreference>(R.string.pref_key_show_sponsored_suggestions).apply {
-            isVisible = context.settings().enableFxSuggest
+            isVisible = false
         }
         requirePreference<SwitchPreference>(R.string.pref_key_show_nonsponsored_suggestions).apply {
             isVisible = context.settings().enableFxSuggest
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index f0eeec863d..032429242b 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -2275,7 +2275,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var enableFxSuggest by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_fxsuggest),
-        default = { FxNimbus.features.fxSuggest.value().enabled },
+        default = { false },
         featureFlag = FeatureFlags.FX_SUGGEST,
     )
 
@@ -2283,7 +2283,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      * Indicates if boosting AMP/wiki suggestions is enabled.
      */
     val boostAmpWikiSuggestions: Boolean
-        get() = FxNimbus.features.fxSuggest.value().boostAmpWiki
+        get() = true
 
     /**
      * Indicates first time engaging with signup
@@ -2300,7 +2300,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     var showSponsoredSuggestions by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_show_sponsored_suggestions),
         default = { enableFxSuggest },
-        featureFlag = FeatureFlags.FX_SUGGEST,
+        featureFlag = false,
     )
 
     /**
@@ -2310,7 +2310,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var showNonSponsoredSuggestions by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_show_nonsponsored_suggestions),
-        default = { enableFxSuggest },
+        default = { false },
         featureFlag = FeatureFlags.FX_SUGGEST,
     )
 
