diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
index b2b40dd764..8042d1c3e7 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
@@ -130,7 +130,7 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_override_user_specified_homepage_sections).apply {
-            isVisible = Config.channel.isNightlyOrDebug
+            isVisible = false
             isChecked = context.settings().overrideUserSpecifiedHomepageSections
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index f0eeec863d..bb0a7fe0f0 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -190,7 +190,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         } else {
             preferences.getBoolean(
                 appContext.getPreferenceKey(R.string.pref_key_customization_bookmarks),
-                homescreenSections[HomeScreenSection.BOOKMARKS] == true,
+                false,
             )
         }
 
@@ -203,7 +203,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         } else {
             preferences.getBoolean(
                 appContext.getPreferenceKey(R.string.pref_key_recent_tabs),
-                homescreenSections[HomeScreenSection.JUMP_BACK_IN] == true,
+                false,
             )
         }
         set(value) {
@@ -239,7 +239,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         } else {
             preferences.getBoolean(
                 appContext.getPreferenceKey(R.string.pref_key_history_metadata_feature),
-                homescreenSections[HomeScreenSection.RECENT_EXPLORATIONS] == true,
+                false,
             )
         }
         set(value) {
@@ -252,13 +252,13 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      * Indicates whether or not the "Synced Tabs" section should be shown on the home screen.
      */
     val showSyncedTabs: Boolean
-        get() = FxNimbus.features.homescreen.value().sectionsEnabled[HomeScreenSection.SYNCED_TABS] == true
+        get() = true
 
     /**
      * Indicates whether or not the "Collections" section should be shown on the home screen.
      */
     val collections: Boolean
-        get() = FxNimbus.features.homescreen.value().sectionsEnabled[HomeScreenSection.COLLECTIONS] == true
+        get() = true
 
     /**
      * Indicates whether or not the homepage header should be shown.
@@ -266,7 +266,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     var showHomepageHeader by lazyFeatureFlagPreference(
         appContext.getPreferenceKey(R.string.pref_key_enable_homepage_header),
         featureFlag = true,
-        default = { homescreenSections[HomeScreenSection.HEADER] == true },
+        default = { true },
     )
 
     /**
@@ -278,7 +278,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         } else {
             preferences.getBoolean(
                 appContext.getPreferenceKey(R.string.pref_key_show_top_sites),
-                homescreenSections[HomeScreenSection.TOP_SITES] == true,
+                true,
             )
         }
 
@@ -314,7 +314,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     val overrideUserSpecifiedHomepageSections by lazyFeatureFlagPreference(
         appContext.getPreferenceKey(R.string.pref_key_override_user_specified_homepage_sections),
-        featureFlag = true,
+        featureFlag = false,
         default = { FxNimbus.features.overrideUserSpecifiedHomepageSections.value().enabled },
     )
 
@@ -514,7 +514,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var showCollectionsPlaceholderOnHome by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_show_collections_placeholder_home),
-        default = true,
+        default = false,
     )
 
     val isCrashReportingEnabled: Boolean
@@ -1381,7 +1381,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var defaultTopSitesAdded by booleanPreference(
         appContext.getPreferenceKey(R.string.default_top_sites_added),
-        default = false,
+        default = true,
     )
 
     var shouldShowSearchSuggestionsInPrivate by booleanPreference(
