diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
index eb55c6630d..f717fe5e01 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt
@@ -367,7 +367,7 @@ class OnboardingFragment : Fragment() {
     }
 
     private fun enableSearchBarCFRForNewUser() {
-        requireContext().settings().shouldShowSearchBarCFR = FxNimbus.features.encourageSearchCfr.value().enabled
+        requireContext().settings().shouldShowSearchBarCFR = false
     }
 
     private fun isNotDefaultBrowser(context: Context) =
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index f0eeec863d..53eac471bd 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -334,7 +334,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     )
 
     val canShowCfr: Boolean
-        get() = (System.currentTimeMillis() - lastCfrShownTimeInMillis) > THREE_DAYS_MS
+        get() = false
 
     var forceEnableZoom by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_accessibility_force_enable_zoom),
@@ -455,7 +455,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var showWallpaperOnboarding by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_wallpapers_onboarding),
-        featureFlag = true,
+        featureFlag = false,
         default = { mr2022Sections[Mr2022Section.WALLPAPERS_SELECTION_TOOL] == true },
     )
 
@@ -494,7 +494,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     var shouldShowMenuBanner by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_show_menu_banner),
         default = { FxNimbus.features.menuRedesign.value().menuBanner },
-        featureFlag = true,
+        featureFlag = false,
     )
 
     var defaultSearchEngineName by stringPreference(
@@ -822,7 +822,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var showFirstTimeTranslation: Boolean by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_show_first_time_translation),
-        default = true,
+        default = false,
     )
 
     /**
@@ -1001,12 +1001,12 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      * This will lead to a performance regression since that function can be expensive to call.
      */
     fun checkIfFenixIsDefaultBrowserOnAppResume(): Boolean {
-        val prefKey = appContext.getPreferenceKey(R.string.pref_key_default_browser)
-        val isDefaultBrowserNow = isDefaultBrowserBlocking()
-        val wasDefaultBrowserOnLastResume =
-            this.preferences.getBoolean(prefKey, isDefaultBrowserNow)
-        this.preferences.edit { putBoolean(prefKey, isDefaultBrowserNow) }
-        return isDefaultBrowserNow && !wasDefaultBrowserOnLastResume
+//        val prefKey = appContext.getPreferenceKey(R.string.pref_key_default_browser)
+//        val isDefaultBrowserNow = isDefaultBrowserBlocking()
+//        val wasDefaultBrowserOnLastResume =
+//            this.preferences.getBoolean(prefKey, isDefaultBrowserNow)
+//        this.preferences.edit { putBoolean(prefKey, isDefaultBrowserNow) }
+        return false
     }
 
     /**
@@ -1015,12 +1015,12 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     fun isDefaultBrowserBlocking(): Boolean {
         val browsers = BrowsersCache.all(appContext)
-        return browsers.isDefaultBrowser
+        return false
     }
 
     var reEngagementNotificationShown by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_re_engagement_notification_shown),
-        default = false,
+        default = true,
     )
 
     /**
@@ -1043,7 +1043,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     var reEngagementNotificationEnabled by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_re_engagement_notification_enabled),
         default = { FxNimbus.features.reEngagementNotification.value().enabled },
-        featureFlag = true,
+        featureFlag = false,
     )
 
     /**
@@ -1051,7 +1051,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     val reEngagementNotificationType: Int
         get() =
-            FxNimbus.features.reEngagementNotification.value().type
+            0
 
     val shouldUseAutoBatteryTheme by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_auto_battery_theme),
@@ -1133,7 +1133,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var shouldShowCookieBannersCFR by lazyFeatureFlagPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_cookie_banners_action_popup),
-        featureFlag = true,
+        featureFlag = false,
         default = { shouldShowCookieBannerUI },
     )
 
@@ -1144,7 +1144,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var hasShownTabSwipeCFR by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_toolbar_has_shown_tab_swipe_cfr),
-        default = false,
+        default = true,
     )
 
     val blockCookiesSelectionInCustomTrackingProtection by stringPreference(
@@ -1449,12 +1449,12 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var userKnowsAboutPwas by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_user_knows_about_pwa),
-        default = false,
+        default = true,
     )
 
     var shouldShowOpenInAppBanner by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_open_in_app_banner),
-        default = true,
+        default = false,
     )
 
     val shouldShowOpenInAppCfr: Boolean
@@ -1462,17 +1462,17 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var shouldShowAutoCloseTabsBanner by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_auto_close_tabs_banner),
-        default = true,
+        default = false,
     )
 
     var shouldShowLockPbmBanner by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_lock_pbm_banner),
-        true,
+        false,
     )
 
     var shouldShowInactiveTabsOnboardingPopup by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_inactive_tabs_popup),
-        default = true,
+        default = false,
     )
 
     /**
@@ -1480,7 +1480,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var hasInactiveTabsAutoCloseDialogBeenDismissed by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_has_inactive_tabs_auto_close_dialog_dismissed),
-        default = false,
+        default = true,
     )
 
     /**
@@ -2290,7 +2290,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var isFirstTimeEngagingWithSignup: Boolean by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_first_time_engage_with_signup),
-        default = true,
+        default = false,
     )
 
     /**
@@ -2468,7 +2468,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     val shouldShowSetAsDefaultPrompt: Boolean
         get() =
-            (System.currentTimeMillis() - lastSetAsDefaultPromptShownTimeInMillis) >
+            false && (System.currentTimeMillis() - lastSetAsDefaultPromptShownTimeInMillis) >
                 DAYS_BETWEEN_DEFAULT_BROWSER_PROMPTS * ONE_DAY_MS &&
                 numberOfSetAsDefaultPromptShownTimes < MAX_NUMBER_OF_DEFAULT_BROWSER_PROMPTS &&
                 coldStartsBetweenSetAsDefaultPrompts >= APP_COLD_STARTS_TO_SHOW_DEFAULT_PROMPT
@@ -2622,7 +2622,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var hasCompletedSetupStepToolbar by booleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_setup_step_toolbar),
-        default = false,
+        default = true,
     )
 
     /**
@@ -2630,7 +2630,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var hasCompletedSetupStepTheme by booleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_setup_step_theme),
-        default = false,
+        default = true,
     )
 
     /**
@@ -2638,7 +2638,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var hasCompletedSetupStepExtensions by booleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_setup_step_extensions),
-        default = false,
+        default = true,
     )
 
     /**
@@ -2646,16 +2646,16 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var hasCompletedSetupStepInstallSearchWidget by booleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_setup_step_install_search_widget),
-        default = false,
+        default = true,
     )
 
     /**
      * Indicates if this is the default browser.
      */
-    var isDefaultBrowser by booleanPreference(
-        key = appContext.getPreferenceKey(R.string.pref_key_default_browser),
-        default = false,
-    )
+    var isDefaultBrowser = true
+//        key = appContext.getPreferenceKey(R.string.pref_key_default_browser),
+//        default = false,
+//    )
 
     /**
      * Indicates if the sponsored tiles are suppressed.
@@ -2675,7 +2675,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
             FxNimbus.features.setupChecklist.value().enabled &&
                     canShowAddSearchWidgetPrompt(AppWidgetManager.getInstance(appContext))
         },
-        featureFlag = true,
+        featureFlag = false,
     )
 
     /**
