diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
index 823b5909fa..91d81c9ce4 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
@@ -525,7 +525,7 @@ class Core(
     val passwordsStorage: SyncableLoginsStorage get() = lazyPasswordsStorage.value
     val autofillStorage: AutofillCreditCardsAddressesStorage get() = lazyAutofillStorage.value
     val domainsAutocompleteProvider: BaseDomainAutocompleteProvider? get() =
-        if (FxNimbus.features.suggestShippedDomains.value().enabled) {
+        if (false) {
             lazyDomainsAutocompleteProvider.value
         } else {
             null
