diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
index 03d80ed014..014a63fa58 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
@@ -772,7 +772,7 @@ class SettingsFragment : PreferenceFragmentCompat() {
     @VisibleForTesting
     internal fun setLinkSharingPreference() {
         with(requirePreference<Preference>(R.string.pref_key_link_sharing)) {
-            isVisible = FxNimbus.features.sentFromFirefox.value().enabled
+            isVisible = false
         }
     }
 
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index f0eeec863d..5592d2df75 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -894,13 +894,13 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var whatsappLinkSharingEnabled by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_link_sharing),
-        featureFlag = true,
+        featureFlag = false,
         default = { FxNimbus.features.sentFromFirefox.value().enabled },
     )
 
     var linkSharingSettingsSnackbarShown by booleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_link_sharing_settings_snackbar),
-        default = false,
+        default = true,
     )
 
     /**
