diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 0bd15b53a6..c29a03539b 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -344,7 +344,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     )
 
     val canShowCfr: Boolean
-        get() = (System.currentTimeMillis() - lastCfrShownTimeInMillis) > THREE_DAYS_MS
+        get() = false
 
     var forceEnableZoom by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_accessibility_force_enable_zoom),
@@ -465,7 +465,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var showWallpaperOnboarding by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_wallpapers_onboarding),
-        featureFlag = true,
+        featureFlag = false,
         default = { mr2022Sections[Mr2022Section.WALLPAPERS_SELECTION_TOOL] == true },
     )
 
@@ -504,7 +504,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     var shouldShowMenuBanner by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_show_menu_banner),
         default = { FxNimbus.features.menuRedesign.value().menuBanner },
-        featureFlag = true,
+        featureFlag = false,
     )
 
     var defaultSearchEngineName by stringPreference(
@@ -832,7 +832,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var showFirstTimeTranslation: Boolean by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_show_first_time_translation),
-        default = true,
+        default = false,
     )
 
     /**
@@ -1053,7 +1053,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     var reEngagementNotificationEnabled by lazyFeatureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_re_engagement_notification_enabled),
         default = { FxNimbus.features.reEngagementNotification.value().enabled },
-        featureFlag = true,
+        featureFlag = false,
     )
 
     /**
@@ -1143,7 +1143,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var shouldShowCookieBannersCFR by lazyFeatureFlagPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_cookie_banners_action_popup),
-        featureFlag = true,
+        featureFlag = false,
         default = { shouldShowCookieBannerUI },
     )
 
@@ -1154,7 +1154,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var hasShownTabSwipeCFR by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_toolbar_has_shown_tab_swipe_cfr),
-        default = false,
+        default = true,
     )
 
     val blockCookiesSelectionInCustomTrackingProtection by stringPreference(
@@ -1459,12 +1459,12 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var userKnowsAboutPwas by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_user_knows_about_pwa),
-        default = false,
+        default = true,
     )
 
     var shouldShowOpenInAppBanner by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_open_in_app_banner),
-        default = true,
+        default = false,
     )
 
     val shouldShowOpenInAppCfr: Boolean
@@ -1472,17 +1472,17 @@ class Settings(private val appContext: Context) : PreferencesHolder {
 
     var shouldShowAutoCloseTabsBanner by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_auto_close_tabs_banner),
-        default = true,
+        default = false,
     )
 
     var shouldShowLockPbmBanner by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_lock_pbm_banner),
-        true,
+        false,
     )
 
     var shouldShowInactiveTabsOnboardingPopup by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_inactive_tabs_popup),
-        default = true,
+        default = false,
     )
 
     /**
@@ -1490,7 +1490,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var hasInactiveTabsAutoCloseDialogBeenDismissed by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_has_inactive_tabs_auto_close_dialog_dismissed),
-        default = false,
+        default = true,
     )
 
     /**
@@ -2300,7 +2300,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
      */
     var isFirstTimeEngagingWithSignup: Boolean by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_first_time_engage_with_signup),
-        default = true,
+        default = false,
     )
 
     /**
@@ -2685,7 +2685,7 @@ class Settings(private val appContext: Context) : PreferencesHolder {
             FxNimbus.features.setupChecklist.value().enabled &&
                     canShowAddSearchWidgetPrompt(AppWidgetManager.getInstance(appContext))
         },
-        featureFlag = true,
+        featureFlag = false,
     )
 
     /**
