diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MainMenu.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MainMenu.kt
index 8da84caa8848a..94d6c190014d5 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MainMenu.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MainMenu.kt
@@ -147,6 +147,8 @@ fun MainMenu(
     showBanner: Boolean,
     isDownloadHighlighted: Boolean,
     webExtensionMenuCount: Int,
+    isPasswordManagerEnabled: Boolean,
+    isSyncActive: Boolean,
     onMoreMenuClick: () -> Unit,
     onCustomizeReaderViewMenuClick: () -> Unit,
     onMozillaAccountButtonClick: () -> Unit,
@@ -297,6 +299,7 @@ fun MainMenu(
 
         LibraryMenuGroup(
             isDownloadHighlighted = isDownloadHighlighted,
+            isPasswordManagerEnabled = isPasswordManagerEnabled,
             onBookmarksMenuClick = onBookmarksMenuClick,
             onHistoryMenuClick = onHistoryMenuClick,
             onDownloadsMenuClick = onDownloadsMenuClick,
@@ -304,12 +307,14 @@ fun MainMenu(
         )
 
         MenuGroup {
-            MozillaAccountMenuItem(
-                account = account,
-                accountState = accountState,
-                isPrivate = isPrivate,
-                onClick = onMozillaAccountButtonClick,
-            )
+            if (isSyncActive) {
+                MozillaAccountMenuItem(
+                    account = account,
+                    accountState = accountState,
+                    isPrivate = isPrivate,
+                    onClick = onMozillaAccountButtonClick,
+                )
+            }
 
             MenuItem(
                 label = stringResource(id = R.string.browser_menu_settings),
@@ -492,6 +497,7 @@ private fun MoreMenuButtonGroup(
 @Suppress("LongMethod")
 private fun LibraryMenuGroup(
     isDownloadHighlighted: Boolean = false,
+    isPasswordManagerEnabled: Boolean = false,
     onBookmarksMenuClick: () -> Unit,
     onHistoryMenuClick: () -> Unit,
     onDownloadsMenuClick: () -> Unit,
@@ -568,16 +574,18 @@ private fun LibraryMenuGroup(
 
         Spacer(Modifier.width(spacerWidth))
 
-        LibraryMenuItem(
-            modifier = Modifier
-                .weight(1f)
-                .fillMaxHeight(),
-            iconRes = iconsR.drawable.mozac_ic_login_24,
-            labelRes = R.string.browser_menu_passwords,
-            shape = rightShape,
-            index = 3,
-            onClick = onPasswordsMenuClick,
-        )
+        if (isPasswordManagerEnabled) {
+            LibraryMenuItem(
+                modifier = Modifier
+                    .weight(1f)
+                    .fillMaxHeight(),
+                iconRes = iconsR.drawable.mozac_ic_login_24,
+                labelRes = R.string.browser_menu_passwords,
+                shape = rightShape,
+                index = 3,
+                onClick = onPasswordsMenuClick,
+            )
+        }
     }
 }
 
@@ -744,6 +752,8 @@ private fun MenuDialogPreview(
                 showBanner = true,
                 isDownloadHighlighted = true,
                 webExtensionMenuCount = 1,
+                isPasswordManagerEnabled = false,
+                isSyncActive = false,
                 onMoreMenuClick = {},
                 onCustomizeReaderViewMenuClick = {},
                 onMozillaAccountButtonClick = {},
@@ -808,6 +818,8 @@ private fun MenuDialogPrivatePreview(
                 showBanner = true,
                 isDownloadHighlighted = true,
                 webExtensionMenuCount = 0,
+                isPasswordManagerEnabled = false,
+                isSyncActive = false,
                 onMoreMenuClick = {},
                 onCustomizeReaderViewMenuClick = {},
                 onMozillaAccountButtonClick = {},
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MoreSettingsSubmenu.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MoreSettingsSubmenu.kt
index 818aea05441e5..07ab90b9e5cba 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MoreSettingsSubmenu.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MoreSettingsSubmenu.kt
@@ -38,6 +38,8 @@ internal fun MoreSettingsSubmenu(
     translationInfo: TranslationInfo,
     showShortcuts: Boolean,
     isAndroidAutomotiveAvailable: Boolean,
+    areCollectionsDisabled: Boolean,
+    isPrintDisabled: Boolean,
     showSummarization: Boolean,
     onWebCompatReporterClick: () -> Unit,
     onSummarizePageClick: () -> Unit,
@@ -75,6 +77,7 @@ internal fun MoreSettingsSubmenu(
             onAddToHomeScreenMenuClick = onAddToHomeScreenMenuClick,
         )
         SaveToCollectionMenuItem(
+            areCollectionsDisabled = areCollectionsDisabled,
             onSaveToCollectionMenuClick = onSaveToCollectionMenuClick,
         )
         OpenInAppMenuItem(
@@ -88,6 +91,7 @@ internal fun MoreSettingsSubmenu(
         )
         PrintMenuItem(
             isAndroidAutomotiveAvailable = isAndroidAutomotiveAvailable,
+            isPrintDisabled = isPrintDisabled,
             onPrintMenuClick = onPrintMenuClick,
         )
     }
@@ -171,13 +175,16 @@ private fun AddToHomeScreenMenuItem(
 
 @Composable
 private fun SaveToCollectionMenuItem(
+    areCollectionsDisabled: Boolean,
     onSaveToCollectionMenuClick: () -> Unit,
 ) {
-    MenuItem(
-        label = stringResource(id = R.string.browser_menu_save_to_collection_2),
-        beforeIconPainter = painterResource(id = iconsR.drawable.mozac_ic_collection_24),
-        onClick = onSaveToCollectionMenuClick,
-    )
+    if (!areCollectionsDisabled) {
+        MenuItem(
+            label = stringResource(id = R.string.browser_menu_save_to_collection_2),
+            beforeIconPainter = painterResource(id = iconsR.drawable.mozac_ic_collection_24),
+            onClick = onSaveToCollectionMenuClick,
+        )
+    }
 }
 
 @Composable
@@ -222,9 +229,10 @@ private fun SaveAsPdfMenuItem(
 @Composable
 private fun PrintMenuItem(
     isAndroidAutomotiveAvailable: Boolean,
+    isPrintDisabled: Boolean,
     onPrintMenuClick: () -> Unit,
 ) {
-    if (!isAndroidAutomotiveAvailable) {
+    if (!isPrintDisabled && !isAndroidAutomotiveAvailable) {
         MenuItem(
             label = stringResource(id = R.string.browser_menu_print_2),
             beforeIconPainter = painterResource(id = iconsR.drawable.mozac_ic_print_24),
@@ -317,6 +325,8 @@ private fun MoreSettingsSubmenuPreview(
                     ),
                     showShortcuts = true,
                     isAndroidAutomotiveAvailable = false,
+                    areCollectionsDisabled = false,
+                    isPrintDisabled = false,
                     showSummarization = true,
                     onWebCompatReporterClick = {},
                     onSummarizePageClick = {},
@@ -363,6 +373,8 @@ private fun MoreSettingsSubmenuDisabledOpenPreview(
                     ),
                     showShortcuts = true,
                     isAndroidAutomotiveAvailable = false,
+                    areCollectionsDisabled = false,
+                    isPrintDisabled = false,
                     showSummarization = false,
                     onWebCompatReporterClick = {},
                     onSummarizePageClick = {},
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/MenuDialogFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/MenuDialogFragment.kt
index 95ded141876d..fadfc00bcda6 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/MenuDialogFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/MenuDialogFragment.kt
@@ -590,7 +590,7 @@ class MenuDialogFragment : BottomSheetDialogFragment() {
                                     accessPoint = args.accesspoint,
                                     account = account,
                                     accountState = accountState,
-                                    showQuitMenu = settings.shouldDeleteBrowsingDataOnQuit,
+                                    showQuitMenu = true,
                                     isBottomToolbar = settings.shouldUseBottomToolbar,
                                     isExpandedToolbarEnabled = settings.shouldUseExpandedToolbar,
                                     isSiteLoading = isSiteLoading,
@@ -610,6 +610,8 @@ class MenuDialogFragment : BottomSheetDialogFragment() {
                                     showBanner = shouldShowMenuBanner && !defaultBrowser,
                                     isDownloadHighlighted = isDownloadHighlighted,
                                     webExtensionMenuCount = webExtensionsCount,
+                                    isPasswordManagerEnabled = settings.shouldPromptToSaveLogins,
+                                    isSyncActive = settings.signedInFxaAccount,
                                     isAllWebExtensionsDisabled = isAllWebExtensionsDisabled,
                                     onMozillaAccountButtonClick = {
                                         store.dispatch(
@@ -708,8 +710,10 @@ class MenuDialogFragment : BottomSheetDialogFragment() {
                                             isWebCompatReporterSupported = isWebCompatReporterSupported,
                                             isOpenInAppMenuHighlighted = isOpenInAppMenuHighlighted,
                                             translationInfo = translationInfo,
-                                            showShortcuts = settings.showTopSitesFeature,
+                                            showShortcuts = settings.showTopSitesFeature && !settings.ironfox.alwaysUsePrivateBrowsing,
                                             isAndroidAutomotiveAvailable = context.isAndroidAutomotiveAvailable(),
+                                            areCollectionsDisabled = !settings.collections,
+                                            isPrintDisabled = !settings.ironfox.printEnabled,
                                             onWebCompatReporterClick = {
                                                 store.dispatch(MenuAction.Navigate.WebCompatReporter)
                                             },
