diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SiteSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SiteSettingsFragment.kt
index 911887b7a2..037e4e8736 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SiteSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SiteSettingsFragment.kt
@@ -104,6 +104,12 @@ class SiteSettingsFragment : PreferenceFragmentCompat() {
             navigateToPhoneFeature(phoneFeature)
             true
         }
+
+        with(settings) {
+            findPreference<Preference>(
+                getPreferenceKey(R.string.pref_key_browser_feature_media_key_system_access),
+            )?.isVisible = emeEnabled
+        }
     }
 
     private fun navigateToPhoneFeature(phoneFeature: PhoneFeature) {
