diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/MarketingPageRemovalSupport.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/MarketingPageRemovalSupport.kt
index fbd12fc813a3..3d0e6f9f5753 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/MarketingPageRemovalSupport.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/MarketingPageRemovalSupport.kt
@@ -67,10 +67,20 @@ class MarketingPageRemovalSupport(
 
 internal fun MutableList<OnboardingPageUiData>.removeIfPageNotReached(index: Int) {
     val marketingIndex = indexOfFirst { it.type == OnboardingPageUiData.Type.MARKETING_DATA }
+    val searchWidgetIndex = indexOfFirst { it.type == OnboardingPageUiData.Type.ADD_SEARCH_WIDGET }
+    val syncIndex = indexOfFirst { it.type == OnboardingPageUiData.Type.SYNC_SIGN_IN }
 
     if (index < marketingIndex) {
         removeAt(marketingIndex)
     }
+
+    if (index < syncIndex) {
+        removeAt(syncIndex)
+    }
+
+    if (index < searchWidgetIndex) {
+        removeAt(searchWidgetIndex)
+    }
 }
 
 internal fun SharedPreferences.flowScopedBooleanPreference(
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt
index 4743fffb0411..ee48ed284bf6 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/about/AboutFragment.kt
@@ -208,11 +208,11 @@ class AboutFragment(
             AboutPageItem(
                 AboutItem.ExternalLink(
                     SUPPORT,
-                    SupportUtils.getSumoURLForTopic(context, SupportUtils.SumoTopic.HELP),
+                    "https://gitlab.com/ironfox-oss/IronFox/-/issues",
                 ),
                 getString(R.string.about_support),
             ),
-            AboutPageItem(
+/*            AboutPageItem(
                 AboutItem.Crashes,
                 getString(R.string.about_crashes),
             ),
@@ -229,7 +229,7 @@ class AboutFragment(
                     SupportUtils.getSumoURLForTopic(context, SupportUtils.SumoTopic.YOUR_RIGHTS),
                 ),
                 getString(R.string.about_know_your_rights),
-            ),
+            ), */
             AboutPageItem(
                 AboutItem.ExternalLink(LICENSING_INFO, ABOUT_LICENSE_URL),
                 getString(R.string.about_licensing_information),
