diff --git a/docshell/base/nsAboutRedirector.cpp b/docshell/base/nsAboutRedirector.cpp
index 83d84537f9..e6faae60b2 100644
--- a/docshell/base/nsAboutRedirector.cpp
+++ b/docshell/base/nsAboutRedirector.cpp
@@ -127,7 +127,7 @@ static const RedirEntry kRedirMap[] = {
      nsIAboutModule::URI_SAFE_FOR_UNTRUSTED_CONTENT |
          nsIAboutModule::URI_CAN_LOAD_IN_CHILD | nsIAboutModule::ALLOW_SCRIPT |
          nsIAboutModule::HIDE_FROM_ABOUTABOUT},
-#if defined(NIGHTLY_BUILD)
+#if !defined(NIGHTLY_BUILD)
     {"inference", "chrome://global/content/aboutInference.html",
      nsIAboutModule::ALLOW_SCRIPT | nsIAboutModule::IS_SECURE_CHROME_UI},
 #endif
diff --git a/docshell/build/components.conf b/docshell/build/components.conf
index 34e06ce332..e8f5e7b055 100644
--- a/docshell/build/components.conf
+++ b/docshell/build/components.conf
@@ -50,7 +50,7 @@ if not defined('MOZ_GLEAN_ANDROID'):
     about_pages.append('glean')
 if buildconfig.substs['MOZ_WIDGET_TOOLKIT'] != 'android' and buildconfig.substs['MOZ_WIDGET_TOOLKIT'] != 'windows':
     about_pages.append('webauthn')
-if defined('NIGHTLY_BUILD'):
+if not defined('NIGHTLY_BUILD'):
     about_pages.append('inference')
 
 Headers = ['/docshell/build/nsDocShellModule.h']
diff --git a/toolkit/components/moz.build b/toolkit/components/moz.build
index dcf865c539..0de04f224f 100644
--- a/toolkit/components/moz.build
+++ b/toolkit/components/moz.build
@@ -145,7 +145,7 @@ if CONFIG["MOZ_BACKGROUNDTASKS"]:
 if CONFIG["MOZ_UNIFFI_FIXTURES"]:
     DIRS += ["uniffi-bindgen-gecko-js/tests"]
 
-if CONFIG["NIGHTLY_BUILD"]:
+if not CONFIG["NIGHTLY_BUILD"]:
     DIRS += ["aboutinference"]
 
 UNIFIED_SOURCES += [
diff --git a/toolkit/locales/jar.mn b/toolkit/locales/jar.mn
index 207e007d2e..72b9c56f72 100644
--- a/toolkit/locales/jar.mn
+++ b/toolkit/locales/jar.mn
@@ -11,7 +11,7 @@
 #ifdef MOZ_LAYOUT_DEBUGGER
   layoutdebug/layoutdebug.ftl                      (../../layout/tools/layout-debug/ui/content/layoutdebug.ftl)
 #endif
-#if defined(NIGHTLY_BUILD)
+#ifndef NIGHTLY_BUILD
   preview/aboutInference.ftl                       (../components/aboutinference/content/aboutInference.ftl)
 #endif
 
