diff --git a/toolkit/components/extensions/NativeManifests.sys.mjs b/toolkit/components/extensions/NativeManifests.sys.mjs
index 5c9cee1b700d..a591196f51e7 100644
--- a/toolkit/components/extensions/NativeManifests.sys.mjs
+++ b/toolkit/components/extensions/NativeManifests.sys.mjs
@@ -35,7 +35,7 @@ export var NativeManifests = {
       let platform = AppConstants.platform;
       if (platform == "win") {
         this._lookup = this._winLookup;
-      } else if (platform == "macosx" || platform == "linux") {
+      } else if (platform == "macosx" || platform == "linux" || platform == "android") {
         let dirs = [
           Services.dirsvc.get("XREUserNativeManifests", Ci.nsIFile).path,
           Services.dirsvc.get("XRESysNativeManifests", Ci.nsIFile).path,
