diff --git a/extensions/pref/autoconfig/src/nsReadConfig.cpp b/extensions/pref/autoconfig/src/nsReadConfig.cpp
index 73da0a726d53..7cbcc7740ffe 100644
--- a/extensions/pref/autoconfig/src/nsReadConfig.cpp
+++ b/extensions/pref/autoconfig/src/nsReadConfig.cpp
@@ -246,7 +246,7 @@ nsresult nsReadConfig::openAndEvaluateJSFile(const char* aFileName,
   nsresult rv;
 
   nsCOMPtr<nsIInputStream> inStr;
-  if (isBinDir) {
+  if (false) {
     nsCOMPtr<nsIFile> jsFile;
 #if defined(MOZ_WIDGET_GTK)
     if (!mozilla::widget::IsRunningUnderFlatpakOrSnap()) {
