diff --git a/toolkit/components/extensions/parent/ext-storage.js b/toolkit/components/extensions/parent/ext-storage.js
index 0d6109c25019..7fa191ecab94 100644
--- a/toolkit/components/extensions/parent/ext-storage.js
+++ b/toolkit/components/extensions/parent/ext-storage.js
@@ -411,6 +411,46 @@ this.storage = class extends ExtensionAPIPersistent {
         managed: {
           async get(keys) {
             enforceNoTemporaryAddon(extension.id);
+            const assetsBootstrapLocation = Services.prefs.getStringPref("browser.ironfox.uBO.assetsBootstrapLocation");
+            const autoCommentFilterTemplate = Services.prefs.getStringPref("browser.ironfox.uBO.autoCommentFilterTemplate");
+            const autoUpdateDelayAfterLaunch = Services.prefs.getStringPref("browser.ironfox.uBO.autoUpdateDelayAfterLaunch");
+            const disableWebAssembly = Services.prefs.getStringPref("browser.ironfox.uBO.disableWebAssembly");
+            const filterAuthorMode = Services.prefs.getStringPref("browser.ironfox.uBO.filterAuthorMode");
+            const uiPopupConfig = Services.prefs.getStringPref("browser.ironfox.uBO.uiPopupConfig");
+            const updateAssetBypassBrowserCache = Services.prefs.getStringPref("browser.ironfox.uBO.updateAssetBypassBrowserCache");
+            if (extension.id == "uBlock0@raymondhill.net") {
+              return {
+                adminSettings: {
+                  assetsBootstrapLocation
+                },
+                advancedSettings: [
+                  [
+                    "autoCommentFilterTemplate",
+                    autoCommentFilterTemplate
+                  ],
+                  [
+                    "autoUpdateDelayAfterLaunch",
+                    autoUpdateDelayAfterLaunch
+                  ],
+                  [
+                    "disableWebAssembly",
+                    disableWebAssembly
+                  ],
+                  [
+                    "filterAuthorMode",
+                    filterAuthorMode
+                  ],
+                  [
+                    "uiPopupConfig",
+                    uiPopupConfig
+                  ],
+                  [
+                    "updateAssetBypassBrowserCache",
+                    updateAssetBypassBrowserCache
+                  ]
+                ]
+              }
+            }
             let data = await getManagedStorageManifestData(extension, context);
             return ExtensionStorage._filterProperties(extension.id, data, keys);
           },
