diff --git a/docshell/build/components.conf b/docshell/build/components.conf
index 76a6380f849c2..589efb14c166b 100644
--- a/docshell/build/components.conf
+++ b/docshell/build/components.conf
@@ -39,7 +39,7 @@ if defined('MOZ_CRASHREPORTER'):
     about_pages.append('crashes')
 if buildconfig.substs['MOZ_WIDGET_TOOLKIT'] != 'android':
     about_pages.append('profiles')
-if buildconfig.substs['MOZ_WIDGET_TOOLKIT'] != 'android':
+if True:
     about_pages.append('translations')
 if buildconfig.substs['MOZ_WIDGET_TOOLKIT'] == 'android':
     about_pages.append('crashcontentjava')
diff --git a/toolkit/modules/ActorManagerParent.sys.mjs b/toolkit/modules/ActorManagerParent.sys.mjs
index e672ab192109f..f0e8f3e664700 100644
--- a/toolkit/modules/ActorManagerParent.sys.mjs
+++ b/toolkit/modules/ActorManagerParent.sys.mjs
@@ -754,7 +754,7 @@ if (AppConstants.platform != "android") {
     messageManagerGroups: ["browsers"],
     allFrames: true,
   };
-
+} else {
   JSWINDOWACTORS.AboutTranslations = {
     parent: {
       esModuleURI: "resource://gre/actors/AboutTranslationsParent.sys.mjs",
@@ -772,8 +772,9 @@ if (AppConstants.platform != "android") {
     },
     matches: ["about:translations"],
     remoteTypes: ["privilegedabout"],
+    enablePreference: "browser.translations.enable",
   };
-
+} if (AppConstants.platform != "android") {
   JSWINDOWACTORS.ColorPicker = {
     parent: {
       esModuleURI: "moz-src:///toolkit/actors/ColorPickerParent.sys.mjs",
