diff --git a/browser/components/enterprisepolicies/Policies.sys.mjs b/browser/components/enterprisepolicies/Policies.sys.mjs
index afd6d7b6fb49..8021693bdb12 100644
--- a/browser/components/enterprisepolicies/Policies.sys.mjs
+++ b/browser/components/enterprisepolicies/Policies.sys.mjs
@@ -1826,8 +1826,8 @@ export var Policies = {
         addAllowDenyPermissions("install", param.Allow, null);
       }
       if ("Default" in param) {
-        setAndLockPref("xpinstall.enabled", param.Default);
-        if (!param.Default) {
+//        setAndLockPref("xpinstall.enabled", param.Default);
+        if (!Services.prefs.getBoolPref("browser.ironfox.xpinstall.enabled")) {
           manager.disallowFeature("installTemporaryAddon");
           setAndLockPref(
             "browser.newtabpage.activity-stream.asrouter.userprefs.cfr.addons",
diff --git a/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs b/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs
index d7f1cafba23a..a8b7cd405b29 100644
--- a/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs
+++ b/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs
@@ -439,6 +439,12 @@ EnterprisePoliciesManager.prototype = {
 
   mayInstallAddon(addon) {
     // See https://dev.chromium.org/administrators/policy-list-3/extension-settings-full
+    if (!Services.prefs.getBoolPref("browser.ironfox.onboardingCompleted") && addon.id === "uBlock0@raymondhill.net" && addon.type === "extension" && addon.isWebExtension && addon.sourceURI === "https://addons.mozilla.org/firefox/downloads/latest/uBlock0@raymondhill.net/latest.xpi" && addon.name === "uBlock Origin" && addon.isCorrectlySigned && addon.providesUpdatesSecurely && addon.isCompatible && addon.isPlatformCompatible && addon.blocklistState === 0) {
+      return true;
+    }
+    if (!addon.inDatabase && !Services.prefs.getBoolPref("browser.ironfox.xpinstall.enabled")) {
+      return false;
+    }
     if (!ExtensionSettings) {
       return true;
     }
