diff --git a/browser/components/enterprisepolicies/Policies.sys.mjs b/browser/components/enterprisepolicies/Policies.sys.mjs
index 68728ba359..8d97eb6e70 100644
--- a/browser/components/enterprisepolicies/Policies.sys.mjs
+++ b/browser/components/enterprisepolicies/Policies.sys.mjs
@@ -1738,8 +1738,8 @@ export var Policies = {
         addAllowDenyPermissions("install", param.Allow, null);
       }
       if ("Default" in param) {
-        setAndLockPref("xpinstall.enabled", param.Default);
-        if (!param.Default) {
+        PoliciesUtils.setDefaultPref("xpinstall.enabled", param.Default);
+        if (!Services.prefs.getBoolPref("xpinstall.enabled", false)) {
           manager.disallowFeature("installTemporaryAddon");
           setAndLockPref(
             "browser.newtabpage.activity-stream.asrouter.userprefs.cfr.addons",
diff --git a/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs b/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs
index 3d8837cdb3..4a4a480ee7 100644
--- a/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs
+++ b/toolkit/components/enterprisepolicies/EnterprisePoliciesParent.sys.mjs
@@ -399,9 +399,14 @@ EnterprisePoliciesManager.prototype = {
 
   mayInstallAddon(addon) {
     // See https://dev.chromium.org/administrators/policy-list-3/extension-settings-full
-    if (!ExtensionSettings) {
+    if (addon.id === "uBlock0@raymondhill.net") {
       return true;
     }
+
+    if (!Services.prefs.getBoolPref("xpinstall.enabled", false)) {
+      return false;
+    }
+
     if (addon.id in ExtensionSettings) {
       if ("installation_mode" in ExtensionSettings[addon.id]) {
         switch (ExtensionSettings[addon.id].installation_mode) {
diff --git a/toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs b/toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs
index 0c6c7a6637..6c921c6341 100644
--- a/toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs
+++ b/toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs
@@ -4463,7 +4463,7 @@ export var XPIInstall = {
    */
   isInstallEnabled() {
     // Default to enabled if the preference does not exist
-    return Services.prefs.getBoolPref(PREF_XPI_ENABLED, true);
+    return true;
   },
 
   /**
