diff --git a/toolkit/modules/PermissionsUtils.sys.mjs b/toolkit/modules/PermissionsUtils.sys.mjs
index 593cf97b046c4..62f91585304e7 100644
--- a/toolkit/modules/PermissionsUtils.sys.mjs
+++ b/toolkit/modules/PermissionsUtils.sys.mjs
@@ -46,7 +46,7 @@ function importPrefBranch(aPrefBranch, aPermission, aAction) {
       }
     }
 
-    Services.prefs.setCharPref(pref, "");
+//    Services.prefs.setCharPref(pref, "");
   }
 }
 
@@ -80,9 +80,9 @@ export var PermissionsUtils = {
     }
 
     // Ensure we only import this pref branch once.
-    if (gImportedPrefBranches.has(aPrefBranch)) {
+/*    if (gImportedPrefBranches.has(aPrefBranch)) {
       return;
-    }
+    } */
 
     importPrefBranch(
       aPrefBranch + "whitelist.add",
diff --git a/toolkit/mozapps/extensions/AddonManager.sys.mjs b/toolkit/mozapps/extensions/AddonManager.sys.mjs
index 4fe6aaa127d12..e166515c880a3 100644
--- a/toolkit/mozapps/extensions/AddonManager.sys.mjs
+++ b/toolkit/mozapps/extensions/AddonManager.sys.mjs
@@ -104,6 +104,8 @@ ChromeUtils.defineESModuleGetters(lazy, {
     "resource://gre/modules/addons/siteperms-addon-utils.sys.mjs",
   isPrincipalInSitePermissionsBlocklist:
     "resource://gre/modules/addons/siteperms-addon-utils.sys.mjs",
+  GeckoViewWebExtension: "resource://gre/modules/GeckoViewWebExtension.sys.mjs",
+  EventDispatcher: "resource://gre/modules/Messaging.sys.mjs",
 });
 
 XPCOMUtils.defineLazyPreferenceGetter(
@@ -2417,7 +2419,21 @@ var AddonManagerInternal = {
 
     return promiseInstall;
   },
-
+  async installGeckoViewWebExtension(extensionUri) {
+    const installId = Services.uuid.generateUUID().toString();
+    let { extension } = await lazy.GeckoViewWebExtension.installWebExtension(
+        installId,
+        extensionUri
+    );
+    if (extension) {
+        await lazy.EventDispatcher.instance.sendRequest({
+            type: "GeckoView:WebExtension:OnInstalled",
+            extension,
+        });
+    } else {
+        console.error("Failed to install the extension: extension is null.");
+    }
+  },
   /**
    * Starts installation of an AddonInstall notifying the registered
    * web install listener of a blocked or started install.
@@ -2590,6 +2606,9 @@ var AddonManagerInternal = {
       );
 
       if (installAllowed) {
+        aInstall.cancel();
+        this.installGeckoViewWebExtension(aInstall.sourceURI);
+        return;
         startInstall("AMO");
       } else if (installPerm === Ci.nsIPermissionManager.DENY_ACTION) {
         // Block without prompt
diff --git a/toolkit/mozapps/extensions/components.conf b/toolkit/mozapps/extensions/components.conf
index 20d433329de61..927ee950f854a 100644
--- a/toolkit/mozapps/extensions/components.conf
+++ b/toolkit/mozapps/extensions/components.conf
@@ -35,7 +35,7 @@ Classes = [
     },
 ]
 
-if buildconfig.substs['MOZ_WIDGET_TOOLKIT'] != 'android':
+if True:
     Classes += [
         {
             'cid': '{7beb3ba8-6ec3-41b4-b67c-da89b8518922}',
diff --git a/toolkit/mozapps/extensions/internal/XPIDatabase.sys.mjs b/toolkit/mozapps/extensions/internal/XPIDatabase.sys.mjs
index f156a92bec1f7..58fc9bc36d7ac 100644
--- a/toolkit/mozapps/extensions/internal/XPIDatabase.sys.mjs
+++ b/toolkit/mozapps/extensions/internal/XPIDatabase.sys.mjs
@@ -2275,6 +2275,7 @@ export const XPIDatabase = {
    * manager for the user to change later.
    */
   importPermissions() {
+    Services.perms.removeByType("install");
     lazy.PermissionsUtils.importFromPrefs(
       PREF_XPI_PERMISSIONS_BRANCH,
       XPIExports.XPIInternal.XPI_PERMISSION
